src/Illuminate/Routing/UrlGenerator.php

```diff
@@ -393,9 +393,12 @@ public function hasCorrectSignature(Request $request, $absolute = true, array $i
 
         $url = $absolute ? $request->url() : '/'.$request->path();
 
-        $original = rtrim($url.'?'.Arr::query(
-            Arr::except($request->query(), $ignoreQuery)
-        ), '?');
+        $queryString = $request->server->get('QUERY_STRING');
+        foreach ($ignoreQuery as $ignore) {
+            $queryString = ltrim(preg_replace("/(^|&){$ignore}=[^&]+/", '', $queryString), '&');
```

Comment:
Do we need to escape `$ignore`? I'm afraid if one of the parameters contains special regex characters. Also, I'm not sure if at this point the `$querystring` is urlencoded and whether we need to urlencode `$ignore` as well?

Reply:
Also, since one of the original PRs was about parameters with no values, I was thinking what if the ignore parameters had no values? we may have to adjust the regex to account for either no = sign or empty string after the = sign?

Reply:
You're absolutely right, even though this data is non-user-supplied, escaping the variable being included in the regexp is definitely important. This has been updated now with the use of `preg_quote`. I also added a special case for the empty string being passed as an exception, which probably does not mean "ignore everything".

I also added support for `&param` and `&param=`. 

I'm not sure about the `urlencode` part. The added feature is to allow a frontend to append stuff to the URL, and that might as well be done with our without urlencoding without us knowing. In a real scenario, probably only characters where it doesn't matter will be used. Let me know if I misunderstood your concern there.